Tests/BuildTests/BuildPlanTests.swift

```diff
@@ -1455,6 +1455,296 @@ final class BuildPlanTests: XCTestCase {
+        ])
+
+        let clangPartOfLib = try result.target(for: "lib").mixedTarget().clangTargetBuildDescription.basicArguments(isCXX: false)
+        #if os(macOS)
+            XCTAssertMatch(clangPartOfLib, [
+                "-fobjc-arc", "-target", "\(defaultTargetTriple)", "-O0",
+                "-DSWIFT_PACKAGE=1", "-DDEBUG=1", "-fblocks", "-fmodules",
+                "-fmodule-name=lib", "-I", "/Pkg/Sources/lib/include", "-I",
+                "/Pkg/Sources/lib", "-I", "/path/to/build/debug/lib.build",
+                "-fmodules-cache-path=/path/to/build/debug/ModuleCache", "-g"
+            ])
+        #elseif os(Linux)
+            XCTAssertMatch(clangPartOfLib, [
+                "-target", "\(defaultTargetTriple)", "-O0",
+                "-DSWIFT_PACKAGE=1", "-DDEBUG=1", "-fblocks", "-fmodules",
+                "-fmodule-name=lib", "-I", "/Pkg/Sources/lib/include", "-I",
+                "/Pkg/Sources/lib", "-I", "/path/to/build/debug/lib.build",
+                "-fmodules-cache-path=/path/to/build/debug/ModuleCache", "-g"
+            ])
+        #endif
+
+        XCTAssertEqual(
+            try result.target(for: "lib").mixedTarget().objects,
+            [
+                buildPath.appending(components: "lib.build", "Foo.swift.o"),
+                buildPath.appending(components: "lib.build", "Bar.m.o"),
+            ]
+        )
+
+        XCTAssertEqual(
+            try result.target(for: "lib").mixedTarget().moduleMap.pathString,
+            "/Pkg/Sources/lib/include/module.modulemap"
+        )
+
+        #if os(macOS)
+            XCTAssertEqual(try result.buildProduct(for: "exe").linkArguments(), [
+                result.plan.buildParameters.toolchain.swiftCompilerPath.pathString,
+                "-L", buildPath.pathString, "-o", buildPath.appending(components: "exe").pathString,
+                "-module-name", "exe", "-emit-executable", "-Xlinker", "-rpath",
+                "-Xlinker", "@loader_path",
+                "@\(buildPath.appending(components: "exe.product", "Objects.LinkFileList"))",
+                "-Xlinker", "-rpath", "-Xlinker", "/fake/path/lib/swift-5.5/macosx",
+                "-target", defaultTargetTriple, "-Xlinker", "-add_ast_path",
+                "-Xlinker", buildPath.appending(component: "exe.swiftmodule").pathString,
+                "-g"
+            ])
+        #elseif os(Linux)
+            XCTAssertEqual(try result.buildProduct(for: "exe").linkArguments(), [
+                result.plan.buildParameters.toolchain.swiftCompilerPath.pathString,
+                "-L", buildPath.pathString, "-o", buildPath.appending(components: "exe").pathString,
+                "-module-name", "exe", "-static-stdlib", "-emit-executable", "-Xlinker", "-rpath=$ORIGIN",
+                "@\(buildPath.appending(components: "exe.product", "Objects.LinkFileList"))",
+                "-target", defaultTargetTriple, "-g"
+            ])
+        #endif
+
+        #if os(macOS)
```

Comment:
Can you clarify in a code comment why exactly the behaviour is different here?

Reply:
TBH, I think I should just remove the observability assertions you highlighted. I'm trying to follow the existing test structure/pattern and copied them from the already existing `BuildPlanTests.testBasicSwiftPackage`, which doesn't explain why this is done.:

https://github.com/apple/swift-package-manager/blob/b96629daa6036d51a1b66728dd78057fa1252052/Tests/BuildTests/BuildPlanTests.swift#L599

https://github.com/apple/swift-package-manager/blob/b96629daa6036d51a1b66728dd78057fa1252052/Tests/BuildTests/BuildPlanTests.swift#L683-L689

Reply:
On macOS, the diagnostic being caught is the following:
```
- [warning]: Swift compiler no longer supports statically linking the Swift libraries. They're included in the OS by default starting with macOS Mojave 10.14.4 beta 3. For macOS Mojave 10.14.3 and earlier, there's an optional Swift library package that can be downloaded from "More Downloads" for Apple Developers at https://developer.apple.com/download/more/
```

Reply:
What do you think the intent may have been here? I could remove the assertion from my tests as it's already being tested in the `BuildPlanTests.testBasicSwiftPackage` test.